adapters/adhese/adhese.go

```diff
@@ -0,0 +1,282 @@
+
+	for k, v := range m {
+		parametersAsString += "/" + k + strings.Join(v, ";")
+	}
+	params := strings.Split(parametersAsString, "/")
+	sort.Strings(params)
+	return strings.Join(params, "/")
+
+}
+
+func extractGdprParameter() string {
+	//const gdprParams = (gdprConsent && gdprConsent.consentString) ? [ 'xt' + gdprConsent.consentString, 'tlall' ] : [];
+	return ""
+}
+
+func extractRefererParameter() string {
+	//const refererParams = (refererInfo && refererInfo.referer) ? [ 'xf' + base64urlEncode(refererInfo.referer) ] : [];
+	return ""
+}
+
+func (a *AdheseAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+
+	var err error
+	var imp = &request.Imp[0]
+	var bidderExt adapters.ExtImpBidder
+
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		errs = append(errs, WrapError("Request could not be parsed as ExtImpBidder due to: "+err.Error()))
+		return nil, errs
+	}
+
+	var params openrtb_ext.ExtImpAdhese
+	if err := json.Unmarshal(bidderExt.Bidder, &params); err != nil {
+		errs = append(errs, WrapError("Request could not be parsed as ExtImpAdhese due to: "+err.Error()))
+		return nil, errs
+	}
+
+	// Validate request
+	if params.Account == "" || params.Location == "" || params.Format == "" {
+		errs = append(errs, WrapError("Request is missing a required parameter (Account, Location and/or Format)"))
+		return nil, errs
+	}
+
+	// Compose url
+	endpointParams := macros.EndpointTemplateParams{Host: "ads-" + params.Account + ".adhese.com"}
+
+	host, err := macros.ResolveMacros(*&a.endpointTemplate, endpointParams)
+	if err != nil {
+		errs = append(errs, WrapError("Could not compose url from template and request account val: "+err.Error()))
+		return nil, errs
+	}
+	complete_url := fmt.Sprintf("%s%s%s%s%s",
+		host,
+		extractSlotParameter(params),
+		extractTargetParameters(params),
+		extractGdprParameter(),
+		extractRefererParameter())
+
+	// If all the requests are invalid, Call to adaptor is skipped
+	if len(request.Imp) == 0 {
+		return nil, errs
+	}
+
+	return []*adapters.RequestData{{
+		Method: "GET",
+		Uri:    complete_url,
+	}}, errs
+}
+
+func (a *AdheseAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{WrapError(fmt.Sprintf("Unexpected status code: %d.", response.StatusCode))}
+	}
+
+	var originArray []openrtb_ext.AdheseOrigin
+	var bidResponse openrtb.BidResponse
+
+	if err := json.Unmarshal(response.Body, &originArray); err != nil {
+		return nil, []error{err, WrapError(fmt.Sprintf("Response %v does not have an Origin.", string(response.Body)))}
+	}
+
+	if originArray[0].Origin == "JERLICIA" || originArray[0].Origin == "DALE" {
```

Comment:
I did not check for sure but I think the response returned by dale is actually closer to the market response rather that jerlicia (I'm quite sure it shouldn't have any properties like `OrderProperty`, `Priority` etc). Was this logic copied from the prebid.js?

Reply:
No, I got that wrong then